components/create-post-page/ScanBarcode.tsx

```diff
@@ -12,33 +11,36 @@ interface Props {
 const ScanBarcode = ({ onDetected, onNoCamera }: Props) => {
   const scannerContainer = useRef(null);
 
-  const updateBarcode = (result: QuaggaJSResultObject) => {
-    if (
-      result.codeResult &&
-      result.codeResult.code?.length === 13 &&
-      // https://www.gs1.org/standards/id-keys/company-prefix
-      (result.codeResult.code.startsWith("978") ||
-        result.codeResult.code.startsWith("979")) &&
-      gtin.validate(result.codeResult.code)
-    ) {
-      Quagga.stop();
-      onDetected(result.codeResult.code);
+  useEffect(() => {
+    if (!scannerContainer.current) {
+      return console.error("Container scanner not found");
     }
-  };
 
-  useEffect(() => {
+    const updateBarcode = (result: QuaggaJSResultObject) => {
+      if (
+        result.codeResult &&
+        result.codeResult.code?.length === 13 &&
+        // https://www.gs1.org/standards/id-keys/company-prefix
+        (result.codeResult.code.startsWith("978") ||
+          result.codeResult.code.startsWith("979")) &&
+        gtin.validate(result.codeResult.code)
+      ) {
+        Quagga.stop();
+        onDetected(result.codeResult.code);
+      }
+    };
+
     Quagga.init(
       {
         inputStream: {
           type: "LiveStream",
-          target: scannerContainer.current ?? undefined,
+          target: scannerContainer.current,
         },
         locator: {
           patchSize: "medium",
           halfSample: true,
         },
         numOfWorkers: 2,
-        frequency: 10,
```

Comment:
Any reason for removing? Might be helpful on lower end devices.

Reply:
I assume slow devices will throttle themselves by just doing the best rate they can? Not sure